File: wmcW/src/views/DashboardView.js
import React from 'react';
import { DashboardViewModel } from '../viewmodels/DashboardViewModel';
import { Sidebar } from './Sidebar';
import { Header } from './Header';
import { DashboardContent } from './DashboardContent';
import { CoursesView } from './CoursesView';
import { MessagingView } from './MessagingView';
import { LiveSessionsView } from './LiveSessionsView';
import { AnalyticsView } from './AnalyticsView';
import { SettingsView } from './SettingsView';

const styles = {
  container: {
    minHeight: "100vh",
    backgroundColor: "#f8fafc",
    fontFamily: "Arial, sans-serif",
    display: "flex"
  },
  mainContent: {
    flex: 1,
    padding: "0",
    overflowY: "auto",
    marginLeft: "280px",
    backgroundColor: "#f8fafc",
    display: "flex",
    flexDirection: "column"
  },
  contentArea: {
    padding: "30px",
    minHeight: "calc(100vh - 100px)",
    backgroundColor: "#f8fafc",
    display: "flex",
    justifyContent: "center",
    alignItems: "flex-start"
  },
  centeredContent: {
    width: "100%",
    maxWidth: "1200px",
    display: "flex",
    flexDirection: "column",
    gap: "30px"
  },
  loadingContainer: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "100vh",
    backgroundColor: "#f8fafc",
    marginLeft: "280px",
    flexDirection: "column",
    gap: "20px"
  },
  loadingText: {
    fontSize: "18px",
    color: "#64748b",
    fontWeight: "500"
  },
  spinner: {
    width: "40px",
    height: "40px",
    border: "4px solid #e2e8f0",
    borderTop: "4px solid #2563eb",
    borderRadius: "50%",
    animation: "spin 1s linear infinite"
  }
};

// Style pour l'animation de spinner
const spinnerStyle = `
  @keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
  }
`;

export const DashboardView = ({ onNavigateToHome }) => {
  const [viewModel] = React.useState(() => new DashboardViewModel());
  const [forceUpdate, setForceUpdate] = React.useState(0);
  const [isLoading, setIsLoading] = React.useState(true);

  // Initialisation des données
  React.useEffect(() => {
    const initializeData = async () => {
      try {
        // Simuler un chargement asynchrone
        await new Promise(resolve => setTimeout(resolve, 1000));
        viewModel.initializeMockData();
        setIsLoading(false);
        setForceUpdate(prev => prev + 1);
      } catch (error) {
        console.error('Error initializing data:', error);
        setIsLoading(false);
      }
    };

    initializeData();
  }, [viewModel]);

  const handleTabChange = (tab) => {
    viewModel.setActiveTab(tab);
    setForceUpdate(prev => prev + 1);
  };

  const handleUpdate = () => {
    setForceUpdate(prev => prev + 1);
  };

  const renderActiveTab = () => {
    if (isLoading) {
      return (
        <div style={styles.loadingContainer}>
          <style>{spinnerStyle}</style>
          <div style={styles.spinner}></div>
          <div style={styles.loadingText}>Loading Dashboard...</div>
        </div>
      );
    }

    const activeContent = (() => {
      switch (viewModel.activeTab) {
        case 'dashboard':
          return (
            <DashboardContent 
              viewModel={viewModel} 
              onUpdate={handleUpdate} 
              key="dashboard-content"
            />
          );
        case 'courses':
          return (
            <CoursesView 
              viewModel={viewModel} 
              onUpdate={handleUpdate} 
              key="courses-view"
            />
          );
        case 'messaging':
          return (
            <MessagingView 
              viewModel={viewModel} 
              onUpdate={handleUpdate} 
              key="messaging-view"
            />
          );
        case 'live':
          return (
            <LiveSessionsView 
              viewModel={viewModel} 
              onUpdate={handleUpdate} 
              key="live-sessions-view"
            />
          );
        case 'analytics':
          return (
            <AnalyticsView 
              viewModel={viewModel} 
              onUpdate={handleUpdate} 
              key="analytics-view"
            />
          );
        case 'settings':
          return (
            <SettingsView 
              viewModel={viewModel} 
              onUpdate={handleUpdate} 
              key="settings-view"
            />
          );
        default:
          return (
            <DashboardContent 
              viewModel={viewModel} 
              onUpdate={handleUpdate} 
              key="default-dashboard"
            />
          );
      }
    })();

    // Retourner le contenu centré
    return (
      <div style={styles.centeredContent}>
        {activeContent}
      </div>
    );
  };

  const handleNavigateHome = () => {
    if (onNavigateToHome) {
      onNavigateToHome();
    }
  };

  return (
    <div style={styles.container}>
      <Sidebar 
        activeTab={viewModel.activeTab}
        onTabChange={handleTabChange}
        onNavigateToHome={handleNavigateHome}
      />
      
      <div style={styles.mainContent}>
        <Header activeTab={viewModel.activeTab} />
        <div style={styles.contentArea}>
          {renderActiveTab()}
        </div>
      </div>
    </div>
  );
};

export default DashboardView;

File: wmcm/lib/views/dashboard_view.dart
import 'package:flutter/material.dart';
import '../viewmodels/dashboard_viewmodel.dart';
import 'components/sidebar.dart';
import 'components/header.dart';
import 'tabs/dashboard_tab.dart';
import 'tabs/courses_tab.dart';
import 'tabs/messaging_tab.dart';
import 'tabs/live_sessions_tab.dart';
import 'tabs/analytics_tab.dart';
import 'tabs/settings_tab.dart';

class DashboardView extends StatefulWidget {
  final VoidCallback onNavigateToHome;

  const DashboardView({super.key, required this.onNavigateToHome});

  @override
  State<DashboardView> createState() => _DashboardViewState();
}

class _DashboardViewState extends State<DashboardView> {
  final DashboardViewModel _viewModel = DashboardViewModel();
  bool _isLoading = true;

  @override
  void initState() {
    super.initState();
    _initializeData();
  }

  void _initializeData() async {
    await Future.delayed(const Duration(seconds: 1));
    _viewModel.initializeMockData();
    setState(() {
      _isLoading = false;
    });
  }

  void _handleTabChange(String tab) {
    setState(() {
      _viewModel.setActiveTab(tab);
    });
  }

  Widget _buildActiveTab() {
    if (_isLoading) {
      return const Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            CircularProgressIndicator(),
            SizedBox(height: 20),
            Text(
              'Loading Dashboard...',
              style: TextStyle(fontSize: 18, color: Colors.grey),
            ),
          ],
        ),
      );
    }

    switch (_viewModel.activeTab) {
      case 'dashboard':
        return DashboardTab(viewModel: _viewModel);
      case 'courses':
        return CoursesTab(viewModel: _viewModel);
      case 'messaging':
        return MessagingTab(viewModel: _viewModel);
      case 'live':
        return LiveSessionsTab(viewModel: _viewModel);
      case 'analytics':
        return AnalyticsTab(viewModel: _viewModel);
      case 'settings':
        return SettingsTab(viewModel: _viewModel);
      default:
        return DashboardTab(viewModel: _viewModel);
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Row(
        children: [
          // Sidebar
          Sidebar(
            activeTab: _viewModel.activeTab,
            onTabChange: _handleTabChange,
            onNavigateToHome: widget.onNavigateToHome,
          ),
          // Main content
          Expanded(
            child: Column(
              children: [
                // Header
                Header(activeTab: _viewModel.activeTab),
                // Content area
                Expanded(
                  child: Padding(
                    padding: const EdgeInsets.all(16.0),
                    child: _buildActiveTab(),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}